internals/cli/cmd_okay.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package cli
+
+import (
+	"fmt"
+
+	"github.com/canonical/go-flags"
+
+	"github.com/canonical/pebble/client"
+)
+
+const cmdOkaySummary = "Acknowledge notices and warnings"
+const cmdOkayDescription = `
+The okay command acknowledges notices up to most recent notice listed with
+'pebble notices'. After executing okay, 'pebble notices' will only list the
+notices that have occurred (or been repeated) more recently.
+
+The okay command also acknowledges warnings up to the most recent warning
+listed with 'pebble warnings'.
+`
+
+type cmdOkay struct {
+	client *client.Client
+
+	Warnings bool `long:"warnings"`
+}
+
+func init() {
+	AddCommand(&CmdInfo{
+		Name:        "okay",
+		Summary:     cmdOkaySummary,
+		Description: cmdOkayDescription,
+		New: func(opts *CmdOptions) flags.Commander {
+			return &cmdOkay{client: opts.Client}
+		},
+		ArgsHelp: map[string]string{
+			"--warnings": "Only acknowledge warnings, not other notices",
+		},
+	})
+}
+
+func (cmd *cmdOkay) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	okayedNotices := false
+	if !cmd.Warnings {
+		state, err := loadNoticesState()
+		if err != nil {
+			return fmt.Errorf("cannot load notices state: %w", err)
+		}
+		if !state.LastListed.IsZero() {
+			okayedNotices = true
+			state.LastOkayed = state.LastListed
+			err = saveNoticesState(state)
+			if err != nil {
+				return fmt.Errorf("cannot save notices state: %w", err)
+			}
+		}
+	}
+
+	last, err := lastWarningTimestamp()
```

Comment:
With this PR, the code is in a funny in-between state, because warnings haven't actually been implemented in terms of notices yet (that's for a future PR). I've tried to make it work, but warnings aren't actually used anywhere yet, so it doesn't matter if this in-between state isn't perfect.

Reply:
If the code doesn't exist yet, what about unifying both inside a single file already, on the notices side since this is the more general concept?  Per earlier note, there will already be more complexity involved due to how pebble isn't a singleton on the system, so we migtht as well do this only once.

Reply:
By both do you mean notices and warnings? I'm not a fan of that, because all our other commands (with the exception of `changes` and `tasks`) are each in their own file, and it's harder to find things if they deviate from that (multiple commands in one file).